src/cache/cloud_fraction.jl

```diff
@@ -94,6 +94,57 @@ NVTX.@annotate function set_cloud_fraction!(
         thermo_params,
     )
 end
+NVTX.@annotate function set_cloud_fraction!(
+    Y,
+    p,
+    ::Union{EquilMoistModel, NonEquilMoistModel},
+    ::DiagnosticEDMFCloud,
+)
+    (; SG_quad, params) = p
+
+    FT = eltype(params)
+    thermo_params = CAP.thermodynamics_params(params)
+    (; ᶜts, ᶜmixing_length, cloud_diagnostics_tuple) = p.precomputed
+    (; ᶜρaʲs, ᶜρʲs, ᶜtsʲs) = p.precomputed
+    (; turbconv_model) = p.atmos
+
+    if p.atmos.call_cloud_diagnostics_per_stage isa CallCloudDiagnosticsPerStage
+        (; ᶜgradᵥ_θ_virt, ᶜgradᵥ_q_tot, ᶜgradᵥ_θ_liq_ice) = p.precomputed
+        thermo_params = CAP.thermodynamics_params(p.params)
+        @. ᶜgradᵥ_θ_virt =
+            ᶜgradᵥ(ᶠinterp(TD.virtual_pottemp(thermo_params, ᶜts)))
+        @. ᶜgradᵥ_q_tot =
+            ᶜgradᵥ(ᶠinterp(TD.total_specific_humidity(thermo_params, ᶜts)))
+        @. ᶜgradᵥ_θ_liq_ice =
+            ᶜgradᵥ(ᶠinterp(TD.liquid_ice_pottemp(thermo_params, ᶜts)))
+    end
```

Comment:
This block should not be needed, as for diagnostic edmf these quantities are already calculated at each stage [here](https://github.com/CliMA/ClimaAtmos.jl/blob/bfabc07266a1a561103d671819115c81cebfac21/src/cache/precomputed_quantities.jl#L484). It might be good to add an assert somewhere so that we only use cloud_model "diagnostic_edmf" when turbconv is "diagnostic_edmfx". (We should unify the name at some point...)

Reply:
Agreed. I don't think this should even be an option for the user. Instead it should be always this way when running with diagnostic edmf. It's just that I wasn't sure how well it will be working in the model and wanted to have it as a temporary option.